File: mobile/app/models/match.ts
import {MeetRequest} from './meet_request'
import {SellRequest} from './sell_request'
import {BuyRequest} from './buy_request'
import {UserPublic} from './user_public'
import {Event} from './event'


export class Match {
    
    public id:number;
    public sell_request:SellRequest;
    public buy_request:BuyRequest;
    public event:Event;
    public seller:UserPublic;
    public buyer:UserPublic;
    public buyer_state:string;
    public seller_state:string;
    public meet_requests:Array<MeetRequest>;
    public state:string;
    
    constructor() {}
    
    public static fromJSON(jsonObj): Match {
        var match:Match = new Match();
        match.id = jsonObj.id;
        match.sell_request = SellRequest.fromJSON(jsonObj.sell_request);
        match.buy_request = BuyRequest.fromJSON(jsonObj.buy_request);
        match.seller = UserPublic.fromJSON(jsonObj.seller);
        match.buyer = UserPublic.fromJSON(jsonObj.buyer);
        match.event = Event.fromJSON(jsonObj.event);
        match.buyer_state = jsonObj.buyer_state;
        match.seller_state = jsonObj.seller_state;
        match.state = jsonObj.state;
        match.meet_requests = new Array<MeetRequest>();
        var mr_list = jsonObj.meet_requests;
        if (mr_list != null) {
            for (var i = 0; i < mr_list.length; i++) {
                match.meet_requests.push(MeetRequest.fromJSON(mr_list[i]));
            }
        }
        return match;
    }
}

File: api/models/match.py
from app import db
from sell_request import SellRequest
from buy_request import BuyRequest
from meet_request import MeetRequest
from user import User
from event import Event

class Match(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    cre_dt = db.Column(db.DateTime)
    up_dt = db.Column(db.DateTime)
    event_id = db.Column(db.Integer, db.ForeignKey(Event.id))
    event = db.relationship(Event)
    sell_request_id = db.Column(db.Integer, db.ForeignKey(SellRequest.id))
    sell_request = db.relationship(SellRequest, uselist=False)
    buy_request_id = db.Column(db.Integer, db.ForeignKey(BuyRequest.id))
    buy_request = db.relationship(BuyRequest, uselist=False)
    seller_id = db.Column(db.Integer, db.ForeignKey(User.id))
    seller = db.relationship(User, uselist=False, foreign_keys=[seller_id])
    buyer_id = db.Column(db.Integer, db.ForeignKey(User.id))
    buyer = db.relationship(User, uselist=False, foreign_keys=[buyer_id])
    seller_state = db.Column(db.String(80))
    buyer_state = db.Column(db.String(80))
    meet_requests = db.relationship(MeetRequest, uselist=True)
    state = db.Column(db.String(80))
